pcap-linux.c

```diff
@@ -3375,10 +3378,19 @@ pcap_setnonblock_linux(pcap_t *handle, int nonblock)
 			 */
 			handlep->timeout = ~handlep->timeout;
 		}
+		if (handlep->poll_breakloop_fd != -1) {
+			/* Close the eventfd; we do not need it in nonblock mode. */
+			close(handlep->poll_breakloop_fd);
+			handlep->poll_breakloop_fd = -1;
+		}
 	} else {
 		if (handlep->timeout < 0) {
 			handlep->timeout = ~handlep->timeout;
 		}
+		if (handlep->poll_breakloop_fd == -1) {
+			/* If we did not have an eventfd, open one now that we are blocking. */
+			handlep->poll_breakloop_fd = eventfd(0, EFD_NONBLOCK);
```

Comment:
This should probably check whether the attempt to (re)open the eventfd fails and, if so, return an error.  (That means the FD should be reopened before changing the timeout, so that, on an error return, non-blocking mode is left on.)

Reply:
I've added this check and a corresponding error, and test that the error value is returned in testprogs/nonblocktest.c.